### PR TITLE
Support environmental properties to set protocol properties #374 + Review properties that can be exposed as managed properties and disallow writing to properties that don't support it #386

### LINKED ISSUE #374: Support environmental properties to set protocol properties
This has been leading to a LOT of confusion with different people trying to configure the stack with properties, but when the S3_PING was added to the stack, they have not added the properties as configurable from the parameters.

When we expose all properties to be set using environmental properties, this won't matter if people are using WildFly, Keycloak, their custom app - it will work the same. Similar to KUBE_PING.

### LINKED ISSUE #386: Review properties that can be exposed as managed properties and disallow writing to properties that don't support it

### PR DESCRIPTION
**Pull request by rhusar**
Resolve #374
Resolve #386

Will be followed up by https://github.com/jgroups-extras/jgroups-aws/issues/390

### CODE REVIEW

**Comment by rhusar** [comment]
Can I get some outside perspective e.g. from @belaban and @tristantarrant ? 

TLDR: result of numerous queries when people were struggling to configure these in Infinispan (especially KC since the stack is opaque to users). Lets expose this configuration as system properties as well, so people don't need to edit the stack and just pass in properties / env variables. Also, have these viewable as management.

n.b. for reference https://github.com/belaban/JGroups/pull/853

**Comment by belaban** [comment]
Not sure what you want from JGroups. It is already possible to set variables in a JGroups config. At runtime, these are searched for in system properties, then env variable, in that order. I'm not sure though if Infinispan overrides this behavior.

**Review by pferraro — APPROVED** [review]
